### ISSUE TITLE
Parse Next or Previous hours

### ISSUE THREAD

**Issue by caiodounis** [briannesbitt/Carbon#1737]
Hello,

It would be nice to use a `parse()` notation like `next 3am` or maybe `coming 3am` to get the coming time it will be `3am`. It would help me in the case when I want to set a countdown. 

For example, instead of writing:

```php
if(Carbon::now() > Carbon::parse('3am')){
	return Carbon::parse('tomorrow 3am');
}
else{
	return Carbon::parse('3am');
}
```

I'd write only:

```php
return Carbon::parse('next 3am');
```

Thanks!

**Comment by kylekatarnls** [comment]
Hi @caiodounis, it's planned for 2.20:
https://github.com/briannesbitt/Carbon/issues/1681

Not yet fully specified but it should allow such operations in one call.

**Comment by kylekatarnls** [comment]
This exact syntax will be available on 2.20, I close this issue, you can review the PR #1746 if you have some remarks about the implementation.
